core/src/main/scala/org/apache/spark/memory/MemoryManager.scala

```diff
@@ -244,17 +245,26 @@ private[spark] abstract class MemoryManager(
    * and then divide it by a factor of safety.
    */
   val pageSizeBytes: Long = {
-    val minPageSize = 1L * 1024 * 1024   // 1MB
-    val maxPageSize = 64L * minPageSize  // 64MB
-    val cores = if (numCores > 0) numCores else Runtime.getRuntime.availableProcessors()
-    // Because of rounding to next power of 2, we may have safetyFactor as 8 in worst case
-    val safetyFactor = 16
-    val maxTungstenMemory: Long = tungstenMemoryMode match {
-      case MemoryMode.ON_HEAP => onHeapExecutionMemoryPool.poolSize
-      case MemoryMode.OFF_HEAP => offHeapExecutionMemoryPool.poolSize
+    val default = Utils.G1HeapRegionSize match {
```

Comment:
lazy val

Reply:
Would lazy do anything here? it's a local inside what's evaluated for a val. I could see declaring it lazily outside this block instead. Then even being lazy doesn't matter

Reply:
It would do:
```
scala> lazy val foo = {
     |   println("Initialized")
     |   1
     | }
foo: Int = <lazy>

scala> "test"
res0: String = test

scala> None.getOrElse(foo)
Initialized
res1: Int = 1
```

But have it outside even better!